### ISSUE TITLE
Remove `encoding` package from dependencies

### ISSUE THREAD

**Issue by MelkorNemesis** [nearform/github-action-check-linked-issues#98]
`node-fetch` version 2.6.7 has been causing us some `esm`/`cjs` compatibility issues. To avoid these issues we had to manually install `encoding` package as a dependency. Newer versions of `node-fetch` package no longer have `encoding` package as just a `peerDependency` (or `dependency` or `devDependency`) and when we learn that the application is using newer version of `node-fetch` we can remove the dependency from `package.json`.

```
$ npm ls node-fetch
github-action-check-linked-issues@1.2.7
└─┬ @actions/github@5.0.3
  └─┬ @octokit/core@3.6.0
    └─┬ @octokit/request@5.6.3
      └── node-fetch@2.6.7
```

More information about the issue:
- https://github.com/nearform/github-action-check-linked-issues/pull/87
- https://github.com/nearform/github-action-check-linked-issues/pull/97

**Comment by denes-fekeshazy** [comment]
Currently the application is still using `@actions/github@5.0.3`, so we can not remove `encoding` as a dependency yet.
Also the problem is that while `node-fetch@2.67` does not list `encoding` as a dependency, it does trying to load the library in one of its functions. At build time it would produce the following code: `module.exports = eval("require")("encoding");`. Which is then failing the `eval("require")` check on this repo.

**Comment by simoneb** [comment]
Correct. We have also put in place a test in CI which checks what ncc generates so that we know if the issue is still there. When we remove the dummy encoding dependency, if that test fails we know that the issue is still there.
